### PR TITLE
added `compare`

### PR DESCRIPTION
**Pull request by ekipan**
I believe it to be ANS Forth correct. I haven't looked at your testing rig but it'd be nice to use the official tests.

This is probably much better done in direct assembly though.

### CODE REVIEW

**Comment by ekipan** [comment]
Bleh, The tests worked in VICE when I tried them from v directly. Cycles again. I do not know what could be wrong.

**Comment by jkotlinski** [comment]
The build archived a screenshots from the test run: 
bild

**Comment by ekipan** [comment]
Probably needs to be in the Makefile? Gimme a sec.

**Comment by ekipan** [comment]
Whoops, maybe I should actually read this thing before committing.

**Comment by jkotlinski** [comment]
The word should probably be described in manual, in "Strings" section.

File: forth/compare.fs [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+if  >r 2drop r> unloop exit  then
+swap 1+ swap 1+ loop  2drop 0 ;
+
+: strcmp ( a1 n1 a2 n2 -- diff )
+rot swap 2dup - >r min memcmp r>
+over if drop else nip then ;
+
+: signum ( n1 -- -1/0/1 )
+dup 0 > if drop 1 exit then
+dup 0< if drop -1 then ;
```
Comment (jkotlinski) [inline forth/compare.fs:12]: `dup 0< if drop -1 then` can probably be replaced by `0<`

**Comment by ekipan** [comment]
Clearly the PR is half-baked. Would you prefer I retract for now?

**Comment by jkotlinski** [comment]
It is fine to keep it open. You can convert it to Draft if you like.

**Comment by ekipan** [comment]
(The force push was an extra blank line in tests.fs I deleted)

**Comment by ekipan** [comment]
Would you prefer file `compare` to be an optional include, as I have here, or try putting it into `base`? I heard the cartridge was tight on bytes.

**Comment by ekipan** [comment]
I wrote this word and thought "hey that seems useful" and submitted on a lark, but really memcmp should be assembler, and I'm sure it already exists in the interpreter somewhere so it would be better to reorganize and expose that, but that's more work than I'm able to do at my level. Thanks for humoring me though.

**Comment by Whammo** [comment]
Would you could Open a Discussion, maybe in Show-and-tell, show your code and document it there? A straight memory compare is a useful thing and offers many opportunities for use-specific optimization. I am certain I would be grateful, and there might even be a half a dozen other people in the world interested.

**Comment by jkotlinski** [comment]
Hey! Sorry for not responding promptly. I had too much real-life things going on lately.

It is true, reuse value would be higher if the word was written efficiently in assembly. I would look at cc65 strcmp/memcmp for inspiration.

I do not think durexForth already has string matching code that is suitable for reuse. It's either too optimized (interpreter.asm) or not particularly good or fast (v.fs).

**Pull request closed without merging**